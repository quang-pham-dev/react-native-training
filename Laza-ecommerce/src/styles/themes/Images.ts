const IMAGES = {
  cover: require('@assets/images/splash.png'),
  iconBack: require('@assets/images/icons/back.png'),
  iconFacebook: require('@assets/images/icons/facebook.png'),
  iconTwitter: require('@assets/images/icons/twitter.png'),
  iconGoogle: require('@assets/images/icons/google.png'),
};

export default IMAGES;
