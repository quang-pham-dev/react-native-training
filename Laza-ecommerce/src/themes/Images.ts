const IMAGES = {
  // icons
  iconBack: require('assets/images/icons/back.webp'),
  iconFacebook: require('assets/images/icons/facebook.webp'),
  iconTwitter: require('assets/images/icons/twitter.webp'),
  iconGoogle: require('assets/images/icons/google.webp'),
  iconMenu: require('assets/images/icons/menu.webp'),
  iconMenuOpen: require('assets/images/icons/menu-open.webp'),
  iconCart: require('assets/images/icons/cart.webp'),
  iconSearch: require('assets/images/icons/search.webp'),
  iconVoice: require('assets/images/icons/voice.webp'),
  iconHome: require('assets/images/icons/home.webp'),
  iconHeart: require('assets/images/icons/heart.webp'),
  iconHeartLiked: require('assets/images/icons/heart-liked.webp'),
  iconBag: require('assets/images/icons/bag.webp'),
  iconWallet: require('assets/images/icons/wallet.webp'),
  iconSun: require('assets/images/icons/sun.webp'),
  iconAvatar: require('assets/images/icons/avatar.webp'),
  iconInfo: require('assets/images/icons/info.webp'),
  iconLogout: require('assets/images/icons/logout.webp'),
  iconBadge: require('assets/images/icons/badge.webp'),
  iconBagDrawer: require('assets/images/icons/bag-drawer.webp'),
  iconHeartDrawer: require('assets/images/icons/heart-drawer.webp'),
  iconWalletDrawer: require('assets/images/icons/wallet-drawer.webp'),
  iconClock: require('assets/images/icons/clock.webp'),
  reviewerAvatar: require('assets/images/icons/reviewer.webp'),
  ratingStart: require('assets/images/icons/start-rate.webp')
};

export default IMAGES;
