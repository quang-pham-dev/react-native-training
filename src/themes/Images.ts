// Icons
const Icons = {
  back: require('@assets/icons/back.webp'),
  facebook: require('@assets/icons/facebook.webp'),
  twitter: require('@assets/icons/twitter.webp'),
  google: require('@assets/icons/google.webp'),
  menu: require('@assets/icons/menu.webp'),
  menuOpen: require('@assets/icons/menu-open.webp'),
  cart: require('@assets/icons/cart.webp'),
  search: require('@assets/icons/search.webp'),
  voice: require('@assets/icons/voice.webp'),
  home: require('@assets/icons/home.webp'),
  heart: require('@assets/icons/heart.webp'),
  heartLiked: require('@assets/icons/heart-liked.webp'),
  bag: require('@assets/icons/bag.webp'),
  wallet: require('@assets/icons/wallet.webp'),
  sun: require('@assets/icons/sun.webp'),
  info: require('@assets/icons/info.webp'),
  logout: require('@assets/icons/logout.webp'),
  badge: require('@assets/icons/badge.webp'),
  bagDrawer: require('@assets/icons/bag-drawer.webp'),
  heartDrawer: require('@assets/icons/heart-drawer.webp'),
  walletDrawer: require('@assets/icons/wallet-drawer.webp'),
  clock: require('@assets/icons/clock.webp'),
  ratingStart: require('@assets/icons/start-rate.webp'),
}

// Images
const Images = {
  splash: require('@assets/images/splash.png'),
}

export {Icons, Images}
