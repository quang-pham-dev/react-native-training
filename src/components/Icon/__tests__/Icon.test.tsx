import React from 'react'
import {render} from '@testing-library/react-native'

import IconStyled from '../Icon.styles'
import {Icons} from '@themes/Images'

const props = {
  type: 'tiny',
  source: Icons.back,
}

describe('Icon', () => {
  test('should render tiny icon correctly', () => {
    const component = render(<IconStyled {...props} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('should render small icon correctly', () => {
    const customProps = {...props, type: 'small'}
    const component = render(<IconStyled {...customProps} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('should render medium icon correctly', () => {
    const customProps = {...props, type: 'medium'}
    const component = render(<IconStyled {...customProps} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('should render large icon correctly', () => {
    const customProps = {...props, type: 'large'}
    const component = render(<IconStyled {...customProps} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
