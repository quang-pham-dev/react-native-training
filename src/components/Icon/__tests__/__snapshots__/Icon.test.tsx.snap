// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render large icon correctly 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/icons/back.webp",
    }
  }
  style={
    Array [
      Object {
        "height": 120,
        "width": 120,
      },
    ]
  }
  type="large"
/>
`;

exports[`Icon should render medium icon correctly 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/icons/back.webp",
    }
  }
  style={
    Array [
      Object {
        "height": 58,
        "width": 58,
      },
    ]
  }
  type="medium"
/>
`;

exports[`Icon should render small icon correctly 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/icons/back.webp",
    }
  }
  style={
    Array [
      Object {
        "height": 24,
        "width": 24,
      },
    ]
  }
  type="small"
/>
`;

exports[`Icon should render tiny icon correctly 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/icons/back.webp",
    }
  }
  style={
    Array [
      Object {
        "height": 10,
        "width": 10,
      },
    ]
  }
  type="tiny"
/>
`;
